src/App.tsx

```diff
@@ -9,36 +9,178 @@ import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { Errors } from './types/Errors';
+import { Comment } from './types/Comment';
+
+import {
+  getComments,
+  getPosts,
+  getUsers,
+} from './api/apiActions';
+
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [isDropDownActive, setIsDropDownActive] = useState(false);
+  const [selectedUsersPosts, setSelectedUsersPosts] = useState<Post[]>([]);
+  const [isLoadingPosts, setIsLoadingPosts] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<Errors>(Errors.noError);
+  const [isLoadingComments, setIsLoadingComments] = useState(false);
+  const [comments, setComments] = useState<Comment[] | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isFormVisible, setIsFormVisible] = useState(false);
```

Comment:
seems some of them are better to be moved to the child components (if possible) since it is complicated to maintain all of the states in one place